modules/notification/notification.go

```diff
@@ -26,7 +27,9 @@ func RegisterNotifier(notifier base.Notifier) {
 
 func init() {
 	RegisterNotifier(ui.NewNotifier())
-	RegisterNotifier(mail.NewNotifier())
+	if setting.Service.EnableNotifyMail {
+		RegisterNotifier(mail.NewNotifier())
+	}
```

Comment:
setting not initialized here,

Reply:
Perhaps this should be moved to an exported function that will be called from `routers.NewServices()` ?

Reply:
Thanks, that seem reasonable.